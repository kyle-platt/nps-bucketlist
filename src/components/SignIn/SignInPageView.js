import React from 'react';
import { Link } from 'react-router-dom';
import * as ROUTES from '../../constants/routes';

const SignInPageView = ({
    email,
    error,
    handleChange,
    handleSubmit,
    isInvalid,
    password,
}) => {
    return (
        <div className="app-background">
            <Link to={ROUTES.LANDING}>back</Link>
            <h1>Let's sign you in</h1>
            <form onSubmit={handleSubmit}>
                <input
                    name="email"
                    value={email}
                    onChange={handleChange}
                    type="text"
                    placeholder="Email Address"
                />
                <input
                    name="password"
                    value={password}
                    onChange={handleChange}
                    type="password"
                    placeholder="Password"
                />
                <Link to={ROUTES.PASSWORD_FORGET}>Forgot Password?</Link>
                <button disabled={isInvalid} type="submit">
                    Sign In
                </button>
                {
                    error && (
                    <p>
                        {error.message}
                    </p>
                )}
            </form>
            <p>
                Don't have an account? <Link to={ROUTES.SIGN_UP}>Sign Up</Link>
            </p>
        </div>
    );
  }

export default SignInPageView;
