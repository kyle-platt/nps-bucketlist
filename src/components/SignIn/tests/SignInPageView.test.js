import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import SignInPageView from '../SignInPageView';

describe('SignInPageView', () => {
  const defaultProps = {
    email: 'info@test.com',
    error: null,
    handleChange: () => {},
    handleSubmit: () => {},
    isInvalid: false,
    password: '123abc',
  };

  it('renders with some data', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <SignInPageView
          {...defaultProps}
        />
      </BrowserRouter> 
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it('renders with an error', () => {
      const tree = renderer.create(
        <BrowserRouter>
          <SignInPageView
            {...defaultProps}
            error={{ message: 'some error' }}
          />
        </BrowserRouter>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
});
