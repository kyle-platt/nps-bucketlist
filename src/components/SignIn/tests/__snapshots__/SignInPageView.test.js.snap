// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with an error 1`] = `
<div
  className="app-background"
>
  <a
    href="/"
    onClick={[Function]}
  >
    back
  </a>
  <h1>
    Let's sign you in
  </h1>
  <form
    onSubmit={[Function]}
  >
    <input
      name="email"
      onChange={[Function]}
      placeholder="Email Address"
      type="text"
      value="info@test.com"
    />
    <input
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value="123abc"
    />
    <a
      href="/pw-forget"
      onClick={[Function]}
    >
      Forgot Password?
    </a>
    <button
      disabled={true}
      type="submit"
    >
      Sign In
    </button>
    <p>
      some error
    </p>
  </form>
  <p>
    Don't have an account? 
    <a
      href="/signup"
      onClick={[Function]}
    >
      Sign Up
    </a>
  </p>
</div>
`;

exports[`renders with some data 1`] = `
<div
  className="app-background"
>
  <a
    href="/"
    onClick={[Function]}
  >
    back
  </a>
  <h1>
    Let's sign you in
  </h1>
  <form
    onSubmit={[Function]}
  >
    <input
      name="email"
      onChange={[Function]}
      placeholder="Email Address"
      type="text"
      value="info@test.com"
    />
    <input
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value="123abc"
    />
    <a
      href="/pw-forget"
      onClick={[Function]}
    >
      Forgot Password?
    </a>
    <button
      disabled={false}
      type="submit"
    >
      Sign In
    </button>
  </form>
  <p>
    Don't have an account? 
    <a
      href="/signup"
      onClick={[Function]}
    >
      Sign Up
    </a>
  </p>
</div>
`;
